kafka/src/main/java/com/metamx/tranquility/kafka/Main.java

```diff
@@ -0,0 +1,149 @@
+import com.metamx.tranquility.kafka.writer.WriterController;
+import io.airlift.airline.Command;
+import io.airlift.airline.Help;
+import io.airlift.airline.HelpOption;
+import io.airlift.airline.Option;
+import io.airlift.airline.SingleCommand;
+import scala.Tuple3;
+import scala.collection.JavaConversions;
+
+import javax.inject.Inject;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * tranquility-kafka main.
+ */
+@Command(name = "tranquility-kafka", description = "Kafka consumer which pushes events to Druid through Tranquility")
+public class Main
+{
+  private static final Logger log = new Logger(Main.class);
+
+  @Inject
+  public HelpOption helpOption;
+
+  @Option(name = {"-f", "-configFile"}, description = "Path to configuration property file")
+  public String propertiesFile;
+
+  public static void main(String[] args) throws Exception
+  {
+    Main main;
+    try {
+      main = SingleCommand.singleCommand(Main.class).parse(args);
+    }
+    catch (Exception e) {
+      log.error(e, "Exception parsing arguments");
+      Help.help(SingleCommand.singleCommand(Main.class).getCommandMetadata());
+      return;
+    }
+
+    if (main.helpOption.showHelpIfRequested()) {
+      return;
+    }
+
+    main.run();
+  }
+
+  public void run() throws InterruptedException
+  {
+    if (propertiesFile == null || propertiesFile.isEmpty()) {
+      helpOption.help = true;
+      helpOption.showHelpIfRequested();
+
+      log.warn("Missing required parameters, aborting.");
+      return;
+    }
+
+    Tuple3<TranquilityKafkaConfig, scala.collection.immutable.Map<String, DataSourceConfig<TranquilityKafkaConfig>>,
+        scala.collection.immutable.Map<String, Object>> configs = null;
+    try (InputStream is = new FileInputStream(propertiesFile)) {
+      configs = ConfigHelper.readConfigYaml(is, TranquilityKafkaConfig.class);
+    }
+    catch (IOException e) {
+      log.error("Could not read config file: %s, aborting.", propertiesFile);
+      Throwables.propagate(e);
+    }
+
+    TranquilityKafkaConfig globalConfig = configs._1();
+    Map<String, DataSourceConfig<TranquilityKafkaConfig>> datasourceConfigs = JavaConversions.mapAsJavaMap(configs._2());
+    Map<String, Object> globalProperties = JavaConversions.mapAsJavaMap(configs._3());
+
+    // find all properties that start with 'kafka.' and pass them on to Kafka
+    final Properties kafkaProperties = new Properties();
+    for (Map.Entry<String, Object> entry : globalProperties.entrySet()) {
+      String key = entry.getKey();
+      log.info("%s = %s", key, globalProperties.get(key).toString());
+      if (key.startsWith("kafka.")) {
+        kafkaProperties.setProperty(key.replaceFirst("kafka.", ""), globalProperties.get(key).toString());
+      }
+    }
+
+    // set the critical Kafka configs again from TranquilityKafkaConfig so it picks up the defaults
```

Comment:
For the first two (kafka.group.id and kafka.zookeeper.connect) it's not possible for the user to specify a different value in both places since they're both based on the same property. For the last one, it could be possible so I can add a check for that one specifically.